packages/utils/core-utils/src/port.ts

```diff
@@ -0,0 +1,41 @@
+import http from 'http';
+
+/**
+ * Check if a port is occupied.
+ * @returns boolean promise that resolves to true if the port is available, false otherwise.
+ */
+export const portOccupied = async (port: number): Promise<void> => {
+  return new Promise<void>((resolve, reject) => {
+    const server = http.createServer().listen(port);
+    server.on('listening', () => {
+      server.close();
+      resolve();
```

Comment:
issue: Not sure if this matters in most cases, but shouldn't this resolve on `server.on('close')` instead so that we're sure the port check won't accidentally block the port?